components/ipfs/ipfs_service.cc

```diff
@@ -1203,7 +1203,7 @@ void IpfsService::OnGatewayValidationComplete(
     const auto final_url = response.final_url();
     const std::string valid_host = base::StringPrintf(
         "%s.ipfs.%s", kGatewayValidationCID, initial_url.host().c_str());
-    success = (response.body() ==
+    success = (response.SerializeBodyToString() ==
                ConvertPlainStringToJsonArray(kGatewayValidationResult)) &&
               (initial_url.host() != final_url.host()) &&
```

Comment:
This migth be refactored so it does not need json in a string

Reply:
IMHO, still it's non-trivial replacement and have to be verified. I don't know how.
I suggest the code owners to fix this after the PR is merged.
@cypt4 WDYT?

Reply:
I think this is just
```
bool CheckGatewayValidationResult(const base::Value& body) {
  return body.is_list() && body.GetList().size() == 1 &&
         body.GetList().front() == kGatewayValidationResult;
}
```

Reply:
No problem to me to put it here if it's okay to the code owners.

Reply:
I believe the best option to add another method to `APIRequestHelper` to process string-only request (without converting it to `base::Value` via `DataDecoder`)